pkg/sql/schemachanger/scbuild/internal/scbuildstmt/drop_sequence.go

```diff
@@ -38,39 +38,28 @@ func DropSequence(b BuildCtx, n *tree.DropSequence) {
 // dropSequence builds targets and transformations using a descriptor.
 func dropSequence(b BuildCtx, seq catalog.TableDescriptor, cascade tree.DropBehavior) {
 	onErrPanic(b.AuthorizationAccessor().CheckPrivilege(b, seq, privilege.DROP))
+	// Add a node to drop the sequence
+	decomposeTableDescToElements(b, seq, scpb.Target_DROP)
 	// Check if there are dependencies.
-	_ = seq.ForeachDependedOnBy(func(dep *descpb.TableDescriptor_Reference) error {
+	scpb.ForEachRelationDependedOnBy(b, func(dep *scpb.RelationDependedOnBy) error {
+		if dep.TableID != seq.GetID() {
+			return nil
+		}
 		if cascade != tree.DropCascade {
-			panic(pgerror.Newf(
+			return pgerror.Newf(
```

Comment:
I find this new pattern where we return instead of panicking quite confusing. My first reaction was to think this was a bug because the return value of  `scpb.ForEachRelationDependedOnBy` is ignored. 

I had to look up the definition of `scpb.ForEachRelationDependedOnBy` to reassure myself that the error will indeed get panicked instead of being quietly ignored. This function and the others like it violate an unspoken style convention which is that `ForEach` functions bubble up the error value returned by the function argument.

Instead, why not have the anonymous fuction passed to `scpb.ForEachRelationDependedOnBy` and friends simply not have a return type at all? Seems more straightforward to me.

Reply:
The intention here was that the iteration could be stopped with an error? @postamar Would a boolean return value make more sense for the call back?

Reply:
Oh I take it back it's already pretty broken, makes more sense to drop the return value completely, since ForEachNode has no way of stopping the iteration